docs/creating-a-new-v3-ea.md

```diff
@@ -0,0 +1,655 @@
+    description: 'Exclusive from param1',
+  },
+} as const
+```
+
+### Cache Key Generator
+
+The `cacheKeyGenerator` method allows the developer to provide custom logic used to generate a unique cache key for each request. The framework's default behavior is to concat the endpoint name with the stringified input parameters such as `test-{"base":"ETH","quote": "USD"}`.
+
+A potential scenario of where a custom cache key generator would be used is if an input parameter is an array or an object. To ensure that a request generates the same unique key every time, the array or object would need to be sorted. An example of this code is shown below.
+
+```typescript
+export const endpoint = new AdapterEndpoint<EndpointTypes>({
+  ...
+  cacheKeyGenerator: (data) => {
+    data.base = (data.base as unknown as string[]).sort()
+    return `test-${JSON.stringify(data)}`
+  },
+  ...
+})
+```
+
+### Custom input validation
+
+The custom input validation method allows the developer to specify custom logic to validate input on top of the built-in validations the framework provides. It returns an `AdapterError` on validation failure, otherwise `undefined` on success. An example is shown below.
+
+```typescript
+export const endpoint = new AdapterEndpoint<EndpointTypes>({
+  ...
+  customInputValidation(req: RequestParams, config: AdapterConfig<typeof customSettings>): AdapterError | undefined => {
+    if (req.value && (req.value < 0 || req.value > 100)) {
+      return new AdapterInputError({
+        statusCode: 400,
+        message: 'Value is less than 0 or greater than 100',
+      })
+    }
+    return
+  },
+  ...
+})
+```
+
+### Price Endpoint
+
+As mentioned in the [Adapter Section](#adapter), `PriceAdapter` requires that one of its endpoints is a `PriceEndpoint`. To help standardize the format of a price feed endpoint, the framework provides helpful built-in types. This would alter some of the examples shown in the previous sub-sections like below.
+
+```typescript
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+type EndpointTypes = {
+  // The PriceEndpointInputParameters type is the most common request params for a price endpoint. {base: string, quote: string}
+  Request: {
+    Params: PriceEndpointInputParameters
+  }
+  ...
+  // The SingleNumberResultResponse type is the most common response format for price endpoints. {result: number, data: { result: number }}
+  Response: SingleNumberResultResponse,
+  ...
+}
+
+export const endpoint = new PriceEndpoint<EndpointTypes>({
+  ...
+})
+```
+
+## Transports
+
+A transport is the method in which the data is retrieved from a data provider for that particular endpoint. As seen in the [Endpoints](#endpoints) section, every endpoint requires one and only one transport. Define this transport in the same file as its associated endpoint. To learn more about specific transports, check the [Transport Types](#transport-types) section.
+
+### Routing Transport
```

Comment:
I don't think we should explain what a transport is here, but rather link to the basics doc and change this to be instead something like "Choosing a transport"